### DISCUSSION TITLE
Use caprover nginx with non docker app

### DISCUSSION THREAD

**Discussion by Tobi-De** [caprover/caprover#2045] [Q&A] [answered]
Hi everyone,

Is it possible to use CapRover's Nginx with non-Docker (or non-CapRover) apps? Let's say I have an app running on my server on a specific port. It was not created with CapRover, but I want (I don't really have a choice anyway) to use CapRover's Nginx instance to expose it with a subdomain.

I know it's a weird use case; I just need to know if it's possible and how.

Thanks :)

**Comment by githubsaturn** [comment, marked as answer]
There are ways to do this! Go to one click apps and deploy `Nginx Reverse Proxy`
![image](https://github.com/caprover/caprover/assets/10582082/cbdbdbd2-842d-4750-a178-e6c4b333e4f2)


For `Upstream address`, use `http://your-server-ip:PORT`. 
`your-server-ip` could be the public IP of your server (your server's local address might also work).

If they still don't work, try this: https://github.com/caprover/caprover/issues/1361

**Reply by Tobi-De** [reply]
Ok thanks, I will try this and get back to you :)

**Reply by Tobi-De** [reply]
OK I got the basis working, thanks @githubsaturn 
Btw how do I add custom nginx configurations ? Would it work If I modify the nginx config of the app ? or is there a special way to add config for this specific nginx container ?

**Reply by githubsaturn** [reply]
Keep in mind that in this setup you have two nginx containers. One is the one that is managed by caprover, and one is the one that you deploy as an app to act as a proxy to your host port. 

You can edit the former using the Edit Nginx configuration button in.the app config. But the one click app nginx cannot be updated on the UI. You can take the image, and modify it using a docker file, and deploy your custom built image.

**Reply by Tobi-De** [reply]
Ok thanks
